code/datums/action.dm

```diff
@@ -688,16 +691,31 @@
 /datum/action/cooldown
 	check_flags = NONE
 	transparent_when_unavailable = FALSE
-	// The default cooldown applied when StartCooldown() is called
+	// The default cooldown applied after the ability ends
 	var/cooldown_time = 0
 	// The actual next time this ability can be used
 	var/next_use_time = 0
+	// The default melee cooldown applied after the ability ends
+	var/melee_cooldown_time
+	// The actual next time the owner of this action can melee
+	var/next_melee_use_time = 0
 	// Whether or not you want the cooldown for the ability to display in text form
 	var/text_cooldown = TRUE
 	// Setting for intercepting clicks before activating the ability
 	var/click_to_activate = FALSE
-	// Shares cooldowns with other cooldown abilities of the same value, not active if null
+	// Shares cooldowns with other abiliies, bitflag
 	var/shared_cooldown
+	// List of prerequisite actions that are used in this sequenced ability, you cannot put other sequenced abilities in this
+	var/list/sequence_actions = list()
+	// List of prerequisite actions that have been initialized
+	var/list/initialized_actions = list()
```

Comment:
These should probably be lazylists given it's more likely they'll not be instantiated than they will